ambassador/ambassador/ir/irhttpmapping.py

```diff
@@ -132,14 +137,26 @@ def __init__(self, ir: 'IR', aconf: Config,
             hdrs.append(Header(":authority", kwargs['host'], kwargs.get('host_regex', False)))
             self.tls_context = self.match_tls_context(kwargs['host'], ir)
 
+        if 'service' in kwargs:
+            svc = Service(ir.logger, kwargs['service'])
+
+            if 'add_linkerd_headers' in kwargs:
+                if kwargs['add_linkerd_headers'] is True: 
+                    add_request_hdrs['l5d-dst-override'] = svc.hostname_port
+            else:
+                if 'add_linkerd_headers' in ir.ambassador_module and ir.ambassador_module.add_linkerd_headers is True:
+                    add_request_hdrs['l5d-dst-override'] = svc.hostname_port
+
```

Comment:
I think this is the logic that should be used in both `Mapping` and `AuthService`.